### PR TITLE
chore: nox doctests only in Python 3.12

### PR DESCRIPTION
**Pull request by EdAbati**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [x] 🐳 Other

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

To be consistent with CI #1189 we should just run doctests in Python 3.12 in `nox`

### CODE REVIEW

File: narwhals/dataframe.py [hunk 1]
```diff
@@ -582,9 +582,9 @@ def write_parquet(self, file: str | Path | BytesIO) -> Any:
 
             We can then pass either pandas, Polars or PyArrow to `func`:
 
-            >>> func(df_pd)
-            >>> func(df_pl)
-            >>> func(df_pa)
+            >>> func(df_pd)  # doctest:+SKIP
+            >>> func(df_pl)  # doctest:+SKIP
+            >>> func(df_pa)  # doctest:+SKIP
```
Comment (EdAbati) [inline narwhals/dataframe.py:587]: these will create a `foo.parquet` locally. If we don't want to skip them, I'll add the file to `.gitignore`

what do you think? :)

Comment (FBruzzesi) [reply narwhals/dataframe.py:587]: I would opt for skipping those! Nice catch 👌

**Review by FBruzzesi — APPROVED** [review]
Awesome! Thanks @EdAbati 😁

DogHighFiveGIF

**Comment by MarcoGorelli** [comment]
thanks both!

**Pull request merged**